src/LineSegment.jl

```diff
@@ -0,0 +1,102 @@
+import Base.∈
+
+export LineSegment
+
+"""
+    LineSegment{N<:Real} <: LazySet{N}
+
+Type that represents a line segment in 2D between two points ``p`` and ``q``.
+
+### Fields
+
+- `p` -- first point
+- `q` -- second point
+"""
+struct LineSegment{N<:Real} <: LazySet{N}
+    p::Vector{N}
```

Comment:
this is a case where static vectors can be useful if heavy use (like arithmetic) over line segments is performed => should we make it an `AbstractVector`?

Reply:
okay :+1: